### PR TITLE
Initialized project base structure

### PR DESCRIPTION
**Pull request by Axisv**
Basic project structure of Meatballs game

### CODE REVIEW

File: app/auth/views.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    return render_template('auth/register.html',
+                           title="Register",
+                           form=form)
+
+
+@auth.route('/login', methods=['GET', 'POST'])
+def login():
+    form = LoginForm()
+    db_user = User.query.filter_by(username=form.username.data).first()
+    if db_user:
```
Comment (cristimihok) [inline app/auth/views.py]: It is possible to combine the two if statements

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+flask
```
Comment (cristimihok) [inline requirements.txt]: Try an alphabetical order

File: .gitignore [hunk 3]
```diff
@@ -52,6 +52,16 @@ coverage.xml
 # Django stuff:
 *.log
 
+# Flask stuff:
+db_repository/
```
Comment (simutoni) [inline .gitignore]: what is the purpose of this? I think on this project this folder is no longer used.

File: app/__init__.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+db = SQLAlchemy(app)
+bcrypt = Bcrypt(app)
+
+login_manager = LoginManager()
+login_manager.init_app(app)
+login_manager.login_view = 'auth.login'
+
+toolbar = DebugToolbarExtension(app)
+
+from .auth import auth as auth_blueprint
```
Comment (simutoni) [inline app/__init__.py]: I think these imports should be placed in the beginning of the file.

File: app/__init__.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+app.register_blueprint(auth_blueprint, url_prefix='/auth')
+
+from .main import main as main_blueprint
```
Comment (simutoni) [inline app/__init__.py]: same story

File: app/auth/__init__.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+from flask import Blueprint
+
+auth = Blueprint('auth', __name__)
```
Comment (simutoni) [inline app/auth/__init__.py]: try to move this on auth views file.

File: app/auth/views.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        if form.validate_on_submit():
+            if bcrypt.check_password_hash(db_user.password, form.password.data):
```
Comment (simutoni) [inline app/auth/views.py]: trey to handle the case when user doesn't provide the right password or username. See also above "if db_user"

File: app/auth/views.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+                login_user(db_user)
+                return redirect(url_for('main.index'))
+
+    return render_template('auth/login.html',
+                           title='Login',
+                           form=form)
+
+
+@auth.route('/logout')
+def logout():
```
Comment (simutoni) [inline app/auth/views.py]: what's happening when the user is not logged in and somehow is going to access the logout route?

File: app/main/__init__.py [hunk 9]
```diff
@@ -0,0 +1,5 @@
+from flask import Blueprint
+
+main = Blueprint('main', __name__)
```
Comment (simutoni) [inline app/main/__init__.py]: try to move this on the views file

File: app/templates/base.html [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    <script type="text/javascript" src="{{ url_for('static', filename='js/app.js') }}"></script>
+</head>
+<body>
+    <header>
+        <nav>
+            <ul>
+
+            <li><a href="/">Home</a></li>
+
+            {% if current_user.is_authenticated == False %}
```
Comment (simutoni) [inline app/templates/base.html]: you can check this like : {% if not current_user.is_authenticated %}

File: app/templates/base.html [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+                <li><a href="{{ url_for('auth.login') }}">Login</a></li>
+                <li><a href="{{ url_for('auth.register') }}">Register</a></li>
+            {% endif %}
+
+            {% if current_user.is_authenticated %}
```
Comment (simutoni) [inline app/templates/base.html]: maybe you can use an else statement here

File: config.py [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+basedir = os.path.abspath(os.path.dirname(__file__))
+
+WTF_CSRF_ENABLED = True
+SECRET_KEY = 'meatballs-game-yonder'
+
+SQLALCHEMY_DATABASE_URI = 'sqlite:///' + os.path.join(basedir, 'app.db')
+SQLALCHEMY_MIGRATE_REPO = os.path.join(basedir, 'db_repository')
+SQLALCHEMY_TRACK_MODIFICATIONS = True
+
+DEBUG = False
```
Comment (simutoni) [inline config.py]: maybe you can change this to True for now.

File: config.py [hunk 13]
```diff
@@ -0,0 +1,12 @@
+import os
+
+basedir = os.path.abspath(os.path.dirname(__file__))
+
+WTF_CSRF_ENABLED = True
```
Comment (EmilLuta) [inline config.py]: This should go to False.

File: models.py [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+
+
+class User(db.Model, UserMixin):
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(50), unique=True, index=True)
+    password = db.Column(db.String(60))
+
+    def __init__(self, username, password):
+        self.username = username
+        self.password = password
```
Comment (EmilLuta) [inline models.py]: Set a representation/ string for this model.

File: db_manage.py [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+            api.create(SQLALCHEMY_MIGRATE_REPO, 'database repository')
+            api.version_control(SQLALCHEMY_DATABASE_URI, SQLALCHEMY_MIGRATE_REPO)
+        else:
+            api.version_control(SQLALCHEMY_DATABASE_URI, SQLALCHEMY_MIGRATE_REPO, api.version(SQLALCHEMY_MIGRATE_REPO))
+
+
+class DbDowngrade(Command):
+    def run(self):
+        database_version = api.db_version(SQLALCHEMY_DATABASE_URI, SQLALCHEMY_MIGRATE_REPO)
+        print('Current database version: ', database_version)
```
Comment (simutoni) [inline db_manage.py]: I think this print is not needed. you have a command for current version on migrate library. Please make the downgrade method to work.

File: manage.py [hunk 16]
```diff
@@ -0,0 +1,15 @@
+from flask.ext.script import Manager
+from app import app
```
Comment (simutoni) [inline manage.py]: please order the imports like:
import a
import b
from x import y
from x import z

File: app/templates/auth/login.html [hunk 17]
```diff
@@ -0,0 +1,19 @@
+{% extends "base.html" %}
+
+{% block content %}
+<section>
+    <h2>Sign In</h2>
+
+    <form  method="post" action="">
```
Comment (simutoni) [inline app/templates/auth/login.html]: please consider testing this :)

File: app/auth/views.py [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+
+    return render_template('auth/register.html',
+                           title="Register",
+                           form=form)
+
+
+@auth.route('/login', methods=['GET', 'POST'])
+def login():
+    form = LoginForm()
+    db_user = User.query.filter_by(username=form.username.data).first()
```
Comment (simutoni) [inline app/auth/views.py]: please add this line only on post.

File: .gitignore (hunk 3 above)
Comment (simutoni) [inline .gitignore]: please rename this in something like db_versions or migrations.

**Pull request merged**